cuda2gcn/src/math.cl

```diff
@@ -0,0 +1,279 @@
+
+//-------- T __nv_erfinv
+//-------- T __nv_erfinvf
+FUNC1(erfinv)
+
+//-------- T __nv_exp
+//-------- T __nv_expf
+FUNC1(exp)
+
+//-------- T __nv_exp10
+//-------- T __nv_exp10f
+FUNC1(exp10)
+
+//-------- T __nv_exp2
+//-------- T __nv_exp2f
+FUNC1(exp2)
+
+//-------- T __nv_expm1
+//-------- T __nv_expm1f
+FUNC1(expm1)
+
+//-------- T __nv_fabs
+//-------- T __nv_fabsf
+FUNC1(fabs)
+
+//-------- T __nv_fdim
+//-------- T __nv_fdimf
+FUNC2(fdim)
+
+//-------- T __nv_floor
+//-------- T __nv_floorf
+FUNC1(floor)
+
+//-------- T __nv_fma
+//-------- T __nv_fmaf
+FUNC3(fma)
+
+//-------- T __nv_fmod
+//-------- T __nv_fmodf
+FUNC2(fmod)
+
+//-------- T __nv_hypot
+//-------- T __nv_hypotf
+FUNC2(hypot)
+
+//-------- T __nv_ilogb
+//-------- T __nv_ilogbf
+FUNC1(ilogb)
+
+//-------- T __nv_ldexp
+//-------- T __nv_ldexpf
+FUNC2(ldexp)
+
+//-------- T __nv_lgamma
+//-------- T __nv_lgammaf
+FUNC1(lgamma)
+
+//-------- T __nv_log
+//-------- T __nv_logf
+FUNC1(log)
+
+//-------- T __nv_log10
+//-------- T __nv_log10f
+FUNC1(log10)
+
+//-------- T __nv_log1p
+//-------- T __nv_log1pf
+FUNC1(log1p)
+
+//-------- T __nv_log2
+//-------- T __nv_log2f
+FUNC1(log2)
+
+//-------- T __nv_logb
+//-------- T __nv_logbf
+FUNC1(logb)
+
+//-------- T __nv_pow
+//-------- T __nv_powf
+FUNC2(pow)
+
+//-------- T __nv_rcbrt
+//-------- T __nv_rcbrtf
+FUNC1(rcbrt)
+
+//-------- T __nv_remainder
+//-------- T __nv_remainderf
+FUNC2(remainder)
+
+//-------- T __nv_nearbyint
+//-------- T __nv_nearbyintf
+FUNC1(nearbyint)
+
+//-------- T __nv_nextafter
+//-------- T __nv_nextafterf
+FUNC2(nextafter)
+
+//-------- T __nv_round
+//-------- T __nv_roundf
+FUNC1(round)
+
+//-------- T __nv_rsqrt
+//-------- T __nv_rsqrtf
+FUNC1(rsqrt)
+
+//-------- T __nv_scalbn
+//-------- T __nv_scalbnf
+FUNC2(scalbn)
+
+//-------- T __nv_sin
+//-------- T __nv_sinf
+FUNC1(sin)
+
+//-------- T __nv_sinh
+//-------- T __nv_sinhf
+FUNC1(sinh)
+
+//-------- T __nv_sinpi
+//-------- T __nv_sinpif
+FUNC1(sinpi)
+
+//-------- T __nv_sqrt
+//-------- T __nv_sqrtf
+FUNC1(sqrt)
+
+//-------- T __nv_tan
+//-------- T __nv_tanf
+FUNC1(tan)
+
+//-------- T __nv_tanh
+//-------- T __nv_tanhf
+FUNC1(tanh)
+
+//-------- T __nv_tgamma
+//-------- T __nv_tgammaf
+FUNC1(tgamma)
+
+//-------- T __nv_trunc
+//-------- T __nv_truncf
+FUNC1(trunc)
+
+//-------- T __nv_abs
+ATTR int __nv_abs(int x) { return abs(x); }
+
+//-------- T __nv_llabs
+ATTR long __nv_llabs(long x) { return abs(x); }
+
+//-------- T __nv_max
+ATTR int __nv_max(int a, int b) { return max(a,b); }
+
+//-------- T __nv_fmax
+ATTR double __nv_fmax(double a, double b) { return max(a,b); }
+
+//-------- T __nv_fmaxf
+ATTR float __nv_fmaxf(float a, float b) { return max(a,b); }
+
+//-------- T __nv_llmax
+ATTR long __nv_llmax(long a, long b) { return max(a,b); }
+
+//-------- T __nv_ullmax
+ATTR ulong __nv_ullmax(ulong a, ulong b) { return max(a,b); }
+
+//-------- T __nv_umax
+ATTR uint __nv_umax(uint a, uint b) { return max(a,b); }
```

Comment:
Consider using a < b ? b : a here and for the other integer functions.  The compiler will do the right thing.